homeassistant/components/brunt/cover.py

```diff
@@ -90,47 +105,82 @@ def request_cover_position(self) -> int | None:
         """
         Return current moving state of cover.
 
         None is unknown, 0 when stopped, 1 when opening, 2 when closing
         """
-        mov = self._state.get("moveState")
+        mov = self.coordinator.data[self.unique_id].moveState
         return int(mov) if mov else None
 
-    def update(self):
-        """Poll the current state of the device."""
-        try:
-            self._state = self._bapi.getState(thingUri=self._thing_uri).get("thing")
-            self._attr_available = True
-        except (TypeError, KeyError, NameError, ValueError) as ex:
-            _LOGGER.error("%s", ex)
-            self._attr_available = False
-        self._attr_is_opening = self.move_state == 1
-        self._attr_is_closing = self.move_state == 2
-        pos = self._state.get("currentPosition")
-        self._attr_current_cover_position = int(pos) if pos else None
-        self._attr_is_closed = self.current_cover_position == CLOSED_POSITION
-        self._attr_extra_state_attributes = {
-            ATTR_ATTRIBUTION: ATTRIBUTION,
-            ATTR_REQUEST_POSITION: self.request_cover_position,
-        }
+    @property
+    def is_opening(self) -> bool | None:
+        """Return if the cover is opening or not."""
+        return self._move_state == 1
+
+    @property
+    def is_closing(self) -> bool | None:
+        """Return if the cover is closing or not."""
+        return self._move_state == 2
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Return true if cover is closed, else False."""
+        return self.current_cover_position == CLOSED_POSITION
 
-    def open_cover(self, **kwargs):
+    async def async_open_cover(self, **kwargs: Any) -> None:
         """Set the cover to the open position."""
-        self._bapi.changeRequestPosition(OPEN_POSITION, thingUri=self._thing_uri)
+        await self._async_update_cover(OPEN_POSITION)
 
-    def close_cover(self, **kwargs):
+    async def async_close_cover(self, **kwargs: Any) -> None:
         """Set the cover to the closed position."""
-        self._bapi.changeRequestPosition(CLOSED_POSITION, thingUri=self._thing_uri)
+        await self._async_update_cover(CLOSED_POSITION)
 
-    def set_cover_position(self, **kwargs):
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
         """Set the cover to a specific position."""
-        self._bapi.changeRequestPosition(
-            kwargs[ATTR_POSITION], thingUri=self._thing_uri
-        )
+        await self._async_update_cover(kwargs[ATTR_POSITION])
+
+    async def _async_update_cover(self, position: int) -> None:
+        """Set the cover to the new position and wait for the update to be reflected."""
+        try:
+            await self._bapi.async_change_request_position(
+                position, thingUri=self._thing.thingUri
+            )
+        except ClientResponseError:
+            _LOGGER.warning("Unable to reposition %s", self._thing.NAME)
```

Comment:
We normally raise `HomeAssistantError` with the message as exception argument if we fail to complete the service handler by another reason than invalid input. The message will be shown to the user in the frontend.